atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/DefaultKnownAbortedTransactions.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.knowledge;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Weigher;
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.transaction.knowledge.AbortedTransactionSoftCache.TransactionSoftCacheStatus;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Set;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+public class DefaultKnownAbortedTransactions implements KnownAbortedTransactions {
+    private final FutileTimestampStore futileTimestampStore;
+
+    @VisibleForTesting
+    static final int MAXIMUM_CACHE_WEIGHT = 100_000;
```

Comment:
discussed offline: let's make this configurable, it's hard to predict what size we'll want